src/cc/ws/SimFederate.java

```
@@ -0,0 +1,121 @@
+package cc.ws;
+
+import java.io.BufferedInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+
+
+public class SimFederate extends HttpServlet implements Runnable
+{
+	protected static final Logger LOGGER = LogManager.getRootLogger();
+
+	private long m_lRetryInterval = 60000L; // registration retry 60 seconds
+	private String m_sAmbassadorAddress;
+	private String m_sCarmaCloudUrl = "";
+	private TimeSource m_oTs;
+
+
+	public SimFederate()
+	{
+	}
+
+
+	@Override
+	public void init()
+	{
+		ServletConfig oConf = getServletConfig();
+
+		String sRetry = oConf.getInitParameter("retry");
+		if (sRetry != null)
+			m_lRetryInterval = Integer.parseInt(sRetry) * 1000L;
+
+		String sAddress = oConf.getInitParameter("ambassador");
+		if (sAddress != null)
+		{
+			try
+			{
+				InetAddress.getByName(sAddress); // validate network address
+				m_sAmbassadorAddress = sAddress;
+				m_oTs = new TimeSource(true);
+
+				String sUrl = oConf.getInitParameter("url");
+				if (sUrl != null)
+					m_sCarmaCloudUrl = sUrl;
+
+				new Thread(this).start(); // begin registration cycle
+			}
+			catch (Exception oEx)
+			{
```

Another empty catch statement. Either log something here or have some behavior here. Catching and doing nothing just effectively silences thrown exceptions.